### PR TITLE
refactor: Adjusting token logo source

### PR DESCRIPTION
**Pull request by Woody810**
### Summary
🧹🚀🎨

Refactored and improved the logic for fetching and displaying token logos in the web app. Extracted the logo URL logic into a utility function `getCurrencyLogoUrls` in `getTokenLogoURL.ts` and used it in `CurrencyLogo.tsx`.

> _We are the masters of the chains_
> _We cache the logos of the tokens_
> _We refactor and we maintain_
> _We are the `CurrencyLogo` rockers_

### Walkthrough
* Extract the logic of getting the currency logo URLs into a utility function `getCurrencyLogoUrls` ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-e4d60339c52e33653f8806738d0332de2e480bbb86597c2288be5ca62a87604fL25-R23), [link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bR32-R44))
  * Use memoization to cache the results based on the currency chain ID and address ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bR32-R44))
  * Use a mapping object to get the correct subdirectory for non-BSC chains ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bR32-R44))
  * Return an array of possible URLs, prioritizing the trust wallet logo over the pancakeswap logo ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bR32-R44))
* Import `Currency` type to the utility file `getTokenLogoURL.ts` ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-f05e9516c94df2fdb153da0702487207e38d053b512fcfa96904de30b9488e6bL3-R3))
* Remove unused imports from the component file `CurrencyLogo.tsx` ([link](https://github.com/pancakeswap/pancake-frontend/pull/6956/files?diff=unified&w=0#diff-e4d60339c52e33653f8806738d0332de2e480bbb86597c2288be5ca62a87604fL3-R5))

Before:
image

After:
image

### CODE REVIEW

File: apps/web/src/utils/getTokenLogoURL.ts [hunk 1]
```diff
@@ -29,4 +29,17 @@ export const getTokenLogoURLByAddress = memoize(
   (address, chainId) => `${chainId}#${address}`,
 )
 
+export const getCurrencyLogoUrls = memoize(
+  (currency?: Currency): string[] => {
+    const chainId = currency?.chainId || ChainId.BSC
+    const tokenAddress = getAddress(currency?.wrapped?.address || '')
+    const trustWalletLogo = getTokenLogoURL(currency?.wrapped)
+    const logoUrl = `https://tokens.pancakeswap.finance/images/${
+      chainId === ChainId.BSC ? '' : `${mapping[chainId]}/`
```
Comment (chef-jojo) [inline apps/web/src/utils/getTokenLogoURL.ts:38]: Mapping is wrong

File: apps/web/src/utils/getTokenLogoURL.ts [hunk 2]
```diff
@@ -29,4 +29,17 @@ export const getTokenLogoURLByAddress = memoize(
   (address, chainId) => `${chainId}#${address}`,
 )
 
+export const getCurrencyLogoUrls = memoize(
+  (currency?: Currency): string[] => {
+    const chainId = currency?.chainId || ChainId.BSC
+    const tokenAddress = getAddress(currency?.wrapped?.address || '')
```
Comment (chef-jojo) [inline apps/web/src/utils/getTokenLogoURL.ts:35]: I remember getAddress will throw error

**Review by chef-jojo — CHANGES_REQUESTED** [review]
Wrapped token logoURI?

File: apps/web/src/utils/getTokenLogoURL.ts (hunk 1 above)
Comment (Woody810) [reply apps/web/src/utils/getTokenLogoURL.ts]: already edited

File: apps/web/src/utils/getTokenLogoURL.ts (hunk 2 above)
Comment (Woody810) [reply apps/web/src/utils/getTokenLogoURL.ts]: already edited

File: apps/web/src/utils/getTokenLogoURL.ts [hunk 3]
```diff
@@ -7,12 +7,25 @@ const mapping = {
...
+  [ChainId.ETHEREUM]: 'eth',
+}
+
 const getTokenLogoURL = memoize(
   (token?: Token) => {
     if (token && mapping[token.chainId]) {
-      return `https://assets-cdn.trustwallet.com/blockchains/${mapping[token.chainId]}/assets/${getAddress(
-        token.address,
-      )}/logo.png`
+      const { chainId } = token
```
Comment (chefjackson) [inline apps/web/src/utils/getTokenLogoURL.ts:18]: Can we try reuse the function here instead?
https://github.com/pancakeswap/pancake-frontend/blob/develop/packages/uikit/src/components/CurrencyLogo/utils.ts#L39

Comment (Woody810) [reply apps/web/src/utils/getTokenLogoURL.ts:18]: Actually, I had already implemented the same function in `apps/web/src/utils/getTokenLogoURL.ts` initially, but JoJo raised the above suggestion, so I made further modifications to the implementation.

**Pull request closed without merging**